src/BloomFilter.cpp

```diff
@@ -18,8 +18,43 @@
 #include <fstream>
 #include <cmath>
 #include <assert.h>
+#include <sys/stat.h>
 #include "BloomFilter.hpp"
 
+
+#define HDR_MAGIC1 0xBF
+#define HDR_MAGIC2 0xAA
+#define HDR_MAGIC3 0xFE
+#define HDR_MAGIC4 0xED
+
+#define VERSION_MAJOR 1
+#define VERSION_MINOR 0
+
+#define INLINE inline
+
+
+// File Headers
+typedef struct LegacyHeader_s {
+    unsigned char e1;
+    unsigned char e2;
+    unsigned char e3;
+    unsigned char e4;
+} LegacyHeader;
+
+typedef struct FileHeader_s {
```

Comment:
I really like this idea however it goes beyond the scope of speeding up the filter and needs some consideration as we would need to do extra work on our backend to support the new structure. I think we will probably do this in the future but let's consider that in a separate issue / PR.